pvanalytics/quality/time.py

```diff
@@ -184,3 +184,123 @@ def shifts_ruptures(event_times, reference_times, period_min=2,
     # localize the shift_amount series to the timezone of the input
     shift_amount = shift_amount.tz_localize(event_times.index.tz)
     return shift_amount != 0, shift_amount
+
+
+def dst_dates(dates, tz):
+    """
+    Get the dates on which daylight savings transitions occur in
+    the timezone `tz`.
+
+    Parameters
+    ----------
+    dates : DatetimeIndex
+        A datetime index with one timestamp per day.
+    tz : str
+        A timezone string (e.g. 'America/Denver' or 'CET')
+
+    Returns
+    -------
+    Series
+        Boolean Series with True on days where daylight savins transitions
+        occur.
+    """
+    midnight = pd.Series(
+        pd.DatetimeIndex(dates.tz_localize(None).date),
+        index=dates
+    )
+    noon = midnight + pd.Timedelta(hours=12)
+    noon = noon.dt.tz_localize(tz)
+    dst_shift = noon.apply(lambda t: t.tzinfo.dst(t).total_seconds() / 3600)
+    return dst_shift.diff().fillna(0) != 0
+
+
+def _has_dst(events, date, window, min_difference):
+    """Return True if `events` appears to have a daylight savings shift
+    on `date`.
+
+    Parameters
+    ----------
+    events : Series
+        Series of timestamps, one per day.
+    date : Timestamp
+        Timestamp corresponding to midnight on the day to check for daylight
+        savings shifts.
+    window : int
```

Comment:
As used, `window` is a Timedelta I think.

Reply:
Yes, it is an int in the API funciton, but a Timedelta here. Thanks, good catch.